scripts/src/java/org/oppia/android/scripts/maven/GenerateLicenseTexts.kt

```diff
@@ -0,0 +1,435 @@
+          }
+          else -> throw Exception(MAVEN_DEPENDENCY_LIST_NOT_UP_TO_DATE)
+        }
+        copyrightLicenseList.add(
+          CopyrightLicense(
+            license.licenseName,
+            licenseLink,
+            licenseText
+          )
+        )
+      }
+      dependencyList.add(
+        Dependency(
+          mavenDependency.artifactName,
+          mavenDependency.artifactVersion,
+          copyrightLicenseList
+        )
+      )
+    }
+    return dependencyList
+  }
+
+  private fun retrieveArtifactsNamesList(dependencyList: List<Dependency>): List<String> {
+    return dependencyList.map { omitVersion(it.name) }
+  }
+
+  private fun retrieveArtifactsVersionsList(
+    dependencyList: List<Dependency>
+  ): List<String> {
+    return dependencyList.map { it.version }
+  }
+
+  private fun writeThirdPartyDependneciesXml(
+    pathToResourceXml: String,
+    dependencyNamesList: List<String>,
+    dependencyVersionsList: List<String>,
+    copyrightLicenseSet: Set<CopyrightLicense>,
+    dependencyList: List<Dependency>
+  ) {
+    val file = File(pathToResourceXml)
+    val docBuilder = DocumentBuilderFactory.newInstance().newDocumentBuilder()
+    val doc = docBuilder.newDocument()
+    val rootResourcesElement = doc.createElement("resources")
+
+    // Create string resources to get dependencies names.
+    writeDependenciesNamesXml(doc, dependencyNamesList, rootResourcesElement)
+
+    // Create string resources to get dependencies versions.
+    writeDependenciesVersionsXml(doc, dependencyVersionsList, rootResourcesElement)
+
+    // Create string resources to get dependencies's license texts and names.
+    writeLicensesNamesAndTextsXml(
+      doc = doc,
+      copyrightLicenseSet = copyrightLicenseSet,
+      rootResourcesElement = rootResourcesElement,
+      dependenciesList = dependencyList
+    )
+
+    doc.appendChild(rootResourcesElement)
+    doc.xmlStandalone = true
+    getTransformer().transform(DOMSource(doc), StreamResult(file))
+  }
+
+  private fun writeDependenciesNamesXml(
+    doc: Document,
+    dependencyNamesList: List<String>,
+    rootResourcesElement: Element
+  ) {
+    // Add all dependencies names as string resources.
+    writeList(
+      itemList = dependencyNamesList,
+      namePrefix = "third_party_dependency_name_",
+      rootResourcesElement = rootResourcesElement,
+      doc = doc
+    )
+
+    // Add all dependencies names in a string-array resource.
+    val stringArrayElement = createArray(
+      arrayTag = "string-array",
+      arrayName = "third_party_dependency_names_array",
+      itemListSize = dependencyNamesList.size,
+      textNodePrefix = "@string/third_party_dependency_name_",
+      doc = doc
+    )
+    rootResourcesElement.appendChild(stringArrayElement)
+  }
+
+  private fun writeDependenciesVersionsXml(
+    doc: Document,
+    dependencyVersionsList: List<String>,
+    rootResourcesElement: Element
+  ) {
+
```

Comment:
Nit: remove extra newline (for consistency). Ditto elsewhere.

Reply:
Done.